File: ads.py
from selenium import webdriver

# location to download file
download_dir = '/Users/alice/Desktop/'

chrome_options = webdriver.ChromeOptions()
prefs = {'download.default_directory' : download_dir}
chrome_options.add_experimental_option('prefs', prefs)
driver = webdriver.Chrome(options=chrome_options)
driver.get('https://www.facebook.com/ads/library/report/')

button = driver.find_element_by_css_selector('.\_7vio')
button.click()

#saves a zip file

File: ads.js
const puppeteer = require("puppeteer");

async function run() {
  const download_path = '/Users/alice/Desktop/'

  const browser = await puppeteer.launch({
    headless: true,
  });
  const page = await browser.newPage();
  await page.setViewport({ width: 1200, height: 1200 });
  await page.goto("https://www.facebook.com/ads/library/report/");
  await page._client.send("Page.setDownloadBehavior", {
    behavior: "allow",
    downloadPath: download_path,
  });

  await page.click('.\_7vio');
  await page.waitFor(5000);

  browser.close();
}

run();
